pkg/eestream/decode.go

```diff
@@ -144,18 +144,33 @@ func (dr *decodedRanger) Size() int64 {
 	return blocks * int64(dr.es.DecodedBlockSize())
 }
 
+type indexReadCloser struct {
+	i int
+	r io.ReadCloser
+}
+
 func (dr *decodedRanger) Range(offset, length int64) io.ReadCloser {
 	// offset and length might not be block-aligned. figure out which
 	// blocks contain this request
 	firstBlock, blockCount := calcEncompassingBlocks(
 		offset, length, dr.es.DecodedBlockSize())
 
 	// go ask for ranges for all those block boundaries
+	// do it parallel to save from network latency
 	readers := make(map[int]io.ReadCloser, len(dr.rrs))
+	result := make(chan indexReadCloser, len(dr.rrs))
 	for i, rr := range dr.rrs {
-		readers[i] = rr.Range(
-			firstBlock*int64(dr.es.EncodedBlockSize()),
-			blockCount*int64(dr.es.EncodedBlockSize()))
+		go func(i int, rr ranger.Ranger) {
+			r := rr.Range(
```

Comment:
does Range do work? i thought this was a fast operation

Reply:
The `Range` of `httpRanger` sends a `GET` request with `Range` header. The returned http body is the `io.ReadCloser` that is returned to the caller. This `GET` request can take some time depending on latency, i.e. seconds.

Reply:
oh you're totally right, okay great work